### ISSUE TITLE
The new distinct function in aggregate transform no longer counts null values. 

### ISSUE THREAD

**Issue by kanitw** [vega/vega#678]
This can be problematic for calculating layout. 

For example,  we should have distinct_MPAA_Rating = 8, not 7, in the spec below.

However, calling `ved.view.data('layout')[0]` gives `{_id: 3919, distinct_MPAA_Rating: 7, width: 21, height: 136}`.

```js
{
  "autosize": "pad",
  "data": [
    {
      "name": "source",
      "url": "data/movies.json",
      "format": {"type": "json","parse": {}},
      "transform": []
    },
    {
      "name": "layout",
      "source": "source",
      "transform": [
        {
          "type": "aggregate",
          "fields": ["MPAA_Rating"],
          "ops": ["distinct"]
        },
        {"type": "formula","as": "width","expr": "21"},
        {
          "type": "formula",
          "as": "height",
          "expr": "(datum[\"distinct_MPAA_Rating\"] + 1) * 17"
        }
      ]
    }
  ],
  "marks": [
    {
      "name": "main",
      "type": "group",
      "description": "A bar chart showing the US population distribution of age groups in 2000.",
      "from": {"data": "layout"},
      "encode": {
        "update": {
          "width": {"field": "width"},
          "height": {"field": "height"}
        }
      },
      "marks": [
        {
          "name": "marks",
          "type": "rect",
          "from": {"data": "source"},
          "encode": {
            "update": {
              "xc": {"value": 10.5,"offset": 1},
              "width": {"value": 20},
              "yc": {"scale": "y","field": "MPAA_Rating"},
              "height": {"value": 16},
              "fill": {"value": "#4682b4"}
            }
          }
        }
      ],
      "scales": [
        {
          "name": "y",
          "type": "point",
          "rangeStep": 17,
          "round": true,
          "padding": 1,
          "domain": {
            "data": "source",
            "field": "MPAA_Rating",
            "sort": true
          }
        }
      ],
      "axes": [
        {
          "scale": "y",
          "grid": false,
          "orient": "left",
          "title": "MPAA_Rating",
          "zindex": 1,
          "encode": {
            "labels": {
              "update": {
                "text": {"signal": "truncate(datum.value, 25)"}
              }
            }
          }
        }
      ]
    }
  ]
}
```

**Comment by kanitw** [comment]
@jheer FYI, this bug is breaking a number of VL examples.  We need this to be fixed (or have a workaround) in order to complete Vega 3 migration.

**Comment by jheer** [comment]
Thanks! I pushed a fix in vega/vega-dataflow@1152ce1.
